WinForm View

TemZZZ opened the pull request:
Лабораторная работа №2

Commit:
style(ConsoleLoader.GetComponent): удалены пустые строки в блоке switch-case

style(ConsoleLoader.PrintComplex): блок if переписан в одну строку

Commit:
refactor(RegexTextBoxLib): переименован из RegularExpressionTextBoxLib

refactor(RegexTextBox): переименован из RegularExpressionTextBox

Commit:
fix(MainForm_Load): в методе остался объект неверного класса RegularExpressionTextBox

- переименован на RegexTextBox

Commit:
style(RegexTextBox.cs): удалены ненужные using

Commit:
feature(RegexTextBox): изменен класс

- удалены конструкторы с параметрами RegexTextBox(string pattern) и public RegexTextBox(Regex regularExpression)
- оставлен один конструктор без параметров, в теле которого вызывается InitializeComponent()
- удален метод Initialize()
- удалена константа defaultText
- удалено поле private readonly Regex _regularExpression
- поле _oldText не инициализируется константой defaultText
- обработчики событий OnTextChanged и OnLostFocus вызывают обработчики базового класса

Commit:
fixture(RegexTextBox): добавлено автосвойство string Pattern

Commit:
feature(RegexTextBox): добавлен метод инициализации объекта Regex

- добавлено private поле Regex _regex
- добавлен private метод CreateRegex

feature(RegexTextBox): свойство Pattern преобразовано в полное свойство

- добавлено private поле _pattern
- в методе set вызывается CreateRegex

Commit:
feature(RegexTextBox): в события OnTextChanged и OnLostFocus добавлена проверка _regex на null

- если _regex == null, то задействовать обработчики базового класса

Commit:
refactor(MainForm): переименованы константы

- doublePattern из doubleNumberPattern
- positiveDoublePattern из positiveDoubleNumberPattern

Commit:
refactor(MainForm): поле positiveDoubleRegex переименовано из positiveDoubleNumberRegularExpression

Commit:
fix(MainForm.MainForm_Load): исправлено создание regexTextBox через несуществующий конструктор с параметром

Commit:
feature(MainForm): удалено поле Regex positiveDoubleRegex

Commit:
feature(Main.MainForm_Load): свойству Pattern объекта regexTextBox присваивается positiveDoublePattern

feature(RegexTextBox.OnTextChanged): реализована логика проверки введенного текста

Commit:
fixture(RegexTextBox.OnLostFocus): обработчик события удален

Commit:
fix(MainForm): исправлен шаблон регулярного выражения positiveDoublePattern

Commit:
fix(MainForm): исправлены шаблоны регулярных выражений doublePattern и positiveDoublePattern

Commit:
fixture(AddRadioComponentForm): добавлены элементы управления

- добавлен valueRegexTextBox
- добавлена кнопка Random Data

Commit:
style(RegexTextBox): добавлены комментарии

Commit:
fixture(MainForm): добавлена обработка клика по кнопке addRadioComponentButton

Commit:
fix(RegexTextBox): добавлена проверка на null в метод set свойства Pattern

Commit:
fixture(AddRadioComponentForm): изменена форма

- FormBorderStyle установлен в FixedSingle
- скрыт значок MaximizeBox

Commit:
fixture(AddRadioComponentForm): реализовано скрытие кнопки Random Data

- кнопка Random Data отображается только если препроцессором установлена переменная TEST

Commit:
refactor(AddRadioComponentForm.valueUnitLabel): переименовано из AddRadioComponentForm.valueLabel

fixture(AddRadioComponentForm): реализована функциональность

- изменяется текст valueUnitLabel в зависимости от выбранной радиокнопки радиокомпонента

Commit:
style(AddRadioComponentForm.RadioButton_CheckedChanged): переименованы константы

- resistorValueUnitText из resistorValueText
- inductorValueUnitText из inductorValueText
- capacitorValueUnitText из capacitorValueText

Commit:
fixture(MainForm): изменена функциональность

- удален #define TEST
- удалены неиспользуемые using
- удален метод MainForm_Load
- переименован метод addRadioComponentButton_Click в AddRadioComponentButton_Click
- создание объекта формы addRadioComponentForm вынесено в конструктор MainForm
- метод AddRadioComponentButton_Click открывает форму addRadioComponentForm как модальное окно

Commit:
style(AddRadioComponentForm): удалены неиспользуемые using

Commit:
feature(AddRadioComponentForm): добавлена возможность закрыть форму кнопкой Отмена

Commit:
feature(MainForm): константы positiveDoublePattern и doublePattern удалены

feature(AddRadioComponentForm): константа positiveDoublePattern добавлена в класс

Commit:
style(AddRadioComponentForm): добавлен комментарий к константе positiveDoublePattern

feature(AddRadioComponentForm): элементу valueRegexTextBox назначен шаблон positiveDoublePattern

Commit:
style(RegexTextBox): добавлен комментарий к свойству Pattern

Commit:
feature(AddRadioComponentForm): добавлена генерация случайных данных нажатием кнопки Random Data

Commit:
fixture(solution Lab1): удален проект Lab1Model

ВАЖНОЕ ИЗМЕНЕНИЕ: проект Lab1Model, написанный под .NetCore 3.1, удален из решения из-за невозможности использования BindingList из экземпляров классов пространства имен Lab1Model

ВАЖНОЕ ИЗМЕНЕНИЕ: проект Lab1Model, переписанный под .NetFramework 4.7.2, добавлен в решение

Commit:
style(ComponentBase): исправлен комментарий к методу get свойства Value

- "знвчение частоты" -> "значение физической величины"

Commit:
fixture(IComponent): добавлены свойства Unit и Type

Commit:
refactor(IRadioComponent): переименован из IComponent

Commit:
refactor(RadioComponentBase): переименован из ComponentBase

Commit:
feature(RadioComponentBase): добавлены абстрактные свойства Unit и Type

Commit:
style(Resistor.CalcImpedance): числовой литерал 0 заменен константой zeroImaginaryPart

Commit:
feature(Resistor): реализованы свойства Unit и Type

Commit:
style(IRadioComponent): исправлен комментарий к свойству Type

- Тип компонента -> Тип радиокомпонента

feature(IRadioComponent): добавлено свойство Quantity

Commit:
feature(RadioComponentBase): добавлено абстрактное свойство Quantity

Commit:
feature(Resistor): реализовано свойство Quantity

Commit:
style(Resistor): сокращены реализации свойств Unit, Type, Quantity

Commit:
feature(Resistor.ToString): изменена реализация с применением свойств Type, Quality, Unit

style(Resistro.ToString): изменен комментарий

Commit:
fixture(Resistor): удален метод ToString

fixture(Inductor): удален метод ToString

fixture(Capacitor): удален метод ToString

fixture(RadioComponentBase): реализован метод ToString

Commit:
feature(MainForm): тип списка radioComponents изменен с List на BindingList

feature(MainForm): изменен модификатор доступа к списку radioComponents с private на internal static

Commit:
style(MainForm): добавлен модификатор доступа к объекту addRadioComponentForm

Commit:
style(MainForm): добавлено ключевое слово readonly к объекту addRadioComponentForm

Commit:
style(AddRadioComponentForm.RadioButton_CheckedChanged): проверка sender на null заменена сопоставлением шаблонов

Commit:
feature(AddRadioComponentForm): добавлен метод проверки строки на соответствие положительному вещественному ToPositiveDouble

Commit:
feature(AddRadioComponentForm): добавлен метод Messager для отображения предупреждений пользователю

Commit:
feature(AddRadioComponentForm): реализовано добавление радиокомпонента в AddRadioComponentButton_Click

Commit:
feature(MainForm): добавлена инициализация списка radioComponents в конструкторе

Commit:
style(MainForm): добавлен комментарий к полю addRadioComponentForm

Commit:
style(AddRadioComponentForm.GenerateRandomDataButton_Click): удалены лишние пустые строки

Commit:
fix(Capacitor): исправлены константы capacitorUnit, capacitorType, capacitorQuantity

Commit:
feature(MainForm): добавлено связывание списка radioComponents с radioComponentsDataGridView в конструкторе класса

Commit:
style(MainForm): у пользователя появилась возможность менять местами столбцы в radioComponentsDataGridView

Commit:
feature(MainForm): добавлено удаление строк кнопкой deleteRadioComponentButton

Commit:
feature(MainForm): пользователь теперь может выделить только всю строку целиком в radioComponentsDataGridView

Commit:
feature(MainForm): добавлен метод FormatRadioComponentsDataGridView

Commit:
feature(MainForm): добавлен вызов метода FormatRadioComponentsDataGridView в конструктор

Commit:
style(MainForm.FormatRadioComponentsDataGridView): добавлен комментарий

Commit:
style(MainForm): изменена геометрия формы

Commit:
feature(Lab1View): добавлен класс SortableBindingList

feature(MainForm): тип списка radioComponents изменен с BindingList на SortableBindingList

Commit:
feature(solution Lab1): добавлен проект PositiveDoubleTextBoxLib

- реализован класс PositiveDoubleTextBox, наследующий RegexTextBox
- класс содержит константу шаблона регулярного выражения положительных вещественных чисел positiveDoublePattern (взят из класса AddRadioComponentForm)
- класс содержит метод ToPositiveDouble (взят из класса AddRadioComponentForm)
- класс содержит метод Messager (взят из класса AddRadioComponentForm)

Commit:
feature(PositiveDoubleTextBox.ToPositiveDouble): messager теперь по умолчанию null

- теперь если не указать messager, сообщения об ошибках не будут передаваться

Commit:
feature(AddRadioComponentForm): valueRegexTextBox заменен на valuePositiveDoubleTextBox

- удалена константа шаблона positiveDoublePattern
- удалено присвоение шаблона элементу valueRegexTextBox (valueRegexTextBox теперь нет на форме)
- удален метод ToPositiveDouble
- удален метод Messager
- в методе GenerateRandomDataButton_Click valueRegexTextBox изменен на valuePositiveDoubleTextBox
- в методе AddRadioComponentButton_Click переписан вызов метода ToPositiveDouble на вызов PositiveDoubleTextBox.ToPositiveDouble без messager
- в методе AddRadioComponentButton_Click valueRegexTextBox изменен на valuePositiveDoubleTextBox

style(AddRadioComponentForm): пересортированы using

Commit:
feature(MainForm): изменена форма

- добавлены кнопки для сохранения в файл/загрузки из файлов радиокомпонентов
- добавлено поле для ввода частоты
- добавлено поле для вывода импеданса
- добавлена кнопка для поиска

Commit:
feature(PositiveDoubleTextBox): добавлен метод GetValue

feature(MainForm): добавлен расчет и вывод импеданса RadioComponentsDataGridView_RowEnter

Commit:
feature(MainForm): добавлен метод ComplexToText

Commit:
feature(MainForm.ComplexToText): добавлен формат строки

Commit:
feature(MainForm.RadioComponentsDataGridView_RowEnter):
добавлена проверка на множественное выделение строк

Commit:
feature(MainForm): добавлены обработчики

- FrequencyPositiveDoubleTextBox_LostFocus
- RadioComponentsDataGridView_RowsAdded

Commit:
style(MainForm): добавлены комментарии к методам

Commit:
feature(Lab1View): изменены стартовые позиции форм

- MainForm запускается по центру экрана
- AddRadioComponentForm запускается по центру MainForm

Commit:
fix(AddRadioComponentForm.AddRadioComponentButton_Click): исправлена ошибка

- пользователю не выводилось сообщение о недопустимости пустого поля значения радиокомпонента, когда изначально у этого поля не было фокуса
- добавлен PositiveDoubleTextBox.Messager в параметры метода PositiveDoubleTextBox.ToPositiveDouble

Commit:
fix(MainForm.RadioComponentsDataGridView_RowEnter): исправлена ошибка

- после удаления последней строки таблицы radioComponentsDataGridView переменная index принимала значение больше, чем элементов в коллекции radioComponents

Commit:
fix(MainForm.DeleteRadioComponentButton_Click): исправлены ошибки, возникающие при удалении строк

- исправлена ошибка при удалении промежуточных строк
- исправлена ошибка при удалении последний строки
- добавлены комментарии в методе, поясняющие ошибки и их исправление

Commit:
style(MainForm): исправлен комментарий в конструкторе

Commit:
style(PositiveDoubleTextBox.cs): удалены неиспользуемые using

Commit:
style(MainForm): изменены Anchor'ы элементов управления

Commit:
feature(MainForm): добавлены openFileDialog и saveFileDialog на форму

feature(MainForm): добавлены пустые обработчики SaveToFileButton_Click и LoadFromFileButton_Click

Commit:
refactor(MainForm.SetupRadioComponentsDataGridView): переименован из MainForm.FormatRadioComponentsDataGridView

Commit:
feature(MainForm): добавлена структура NameHeaderTextPair

- структура используется для редактирования внешнего вида таблицы DataGridView

feature(MainForm.SetupRadioComponentsDataGridView): метод переписан с применением структур NameHeaderTextPair

style(MainForm.SetupRadioComponentsDataGridView): обновлен комментарий

Commit:
style(MainForm.cs): блоки if оформлены по MSDN

Commit:
style(RadioComponentBase.cs): код оформлен в соответствие с MSDN

Commit:
style(AddRadioComponentForm.cs): код оформлен в соответствие с MSDN

Commit:
style(SortableBindingList.cs): код оформлен в соответствие с MSDN

Commit:
style(PositiveDoubleTextBox.cs): код оформлен в соответствие с MSDN

Commit:
style(RegexTextBox.cs): добавлены XML-комментарии, код оформлен по MSDN

Commit:
style(PositiveDoubleTextBox.cs): добавлены XML-комментарии к конструктору, к OnLostFocus

Commit:
style(RegexTextBox.cs): исправлен комментарий к конструктору

Commit:
style(MainForm.cs): добавлены XML-комментарии к классу и конструктору

Commit:
style(AddRadioComponentForm.cs): добавлены XML-комментарии

Commit:
feature(RadioComponentBase): добавлены атрибуты для возможности сериализации объектов Resistor, Inductor и Capacitor

Commit:
style(RadioComponentBase): удален комментарий про атрибуты сериализации

- этот комментарий не позволял корректно отбражать описание класса в VisualStudio

iKalentii commented on Lab1Model_DotNetFramework472/Lab1Model/Lab1Model/Capacitor.cs:
```diff
@@ -0,0 +1,43 @@
+			/// Создается экземпляр класса <see cref="Capacitor"/>
+			/// </summary>
+			/// <param name="value">Значение емкости в фарадах</param>
+			public Capacitor(double value) : base(value) { }
```
XML комменты

TemZZZ replied:
К чему относится замечание? К комментарию к конструктору Capacitor или к следующему за ним методу? Если к следующему, то при наведении мышью отображается описание из базового класса:
![image](https://user-images.githubusercontent.com/63415775/81652245-a3d90680-945d-11ea-81e5-33d66532df3d.png)

iKalentii replied:
К методу CalcImpedance. Если в базовом стоят комменты - тогда тут надо inheritdocs повешать да и всё.

iKalentii commented on Lab1Model_DotNetFramework472/Lab1Model/Lab1Model/RadioComponentBase.cs:
```diff
@@ -0,0 +1,99 @@
+
+            set
+            {
+                /// <summary>
```
set и get - считай также методами, поэтому внутри не надо писать XML комментарии

iKalentii commented on Lab1Model_DotNetFramework472/Lab1Model/Lab1Model/RadioComponentBase.cs:
```diff
@@ -0,0 +1,99 @@
+            // то бросить исключение ArgumentOutOfRangeException
+
+            if (freq < 0)
+                throw new ArgumentOutOfRangeException(
```
Скобочки

iKalentii commented on Lab1Model_DotNetFramework472/Lab1Model/Lab1Model/RadioComponentBase.cs:
```diff
@@ -0,0 +1,99 @@
+        /// значения частоты</exception>
+        public Complex GetImpedance(double freq)
+        {
+            // Если значение частоты меньше нуля,
```
Из кода понятно - что делается, комментарий не нужен

iKalentii commented on Lab1Model_DotNetFramework472/Lab1Model/Lab1Model/RadioComponentBase.cs:
```diff
@@ -0,0 +1,99 @@
+            return $"Тип: {Type}; {Quantity} = {Value} {Unit}";
+        }
+
+        public abstract string Unit { get; }
```
XML

TemZZZ replied:
Здесь тоже уже есть комментарии у свойств (как в первом замечании):
![image](https://user-images.githubusercontent.com/63415775/81660948-a0497d80-9465-11ea-8fca-0b8204785a20.png)
Нужно дублировать как в базовом классе или интерфейсе?

iKalentii replied:
Выше отписался, вешай <inheritdocs>

iKalentii commented on Lab1View/AddRadioComponentForm.cs:
```diff
@@ -1,21 +1,151 @@
+            const string capacitorValueUnitText = "Емкость, Ф";
+
+            if (selectedRadioButton == resistorRadioButton)
+                valueUnitLabel.Text = resistorValueUnitText;
```
Скобочки.

iKalentii commented on Lab1View/MainForm.cs:
```diff
@@ -1,28 +1,263 @@
+        /// Список радиокомпонентов
+        /// </summary>
+        internal static
+            SortableBindingList<RadioComponentBase> radioComponents;
```
RSDN

iKalentii replied:
Также нет смысла делать статическим

TemZZZ replied:
Если поле RadioComponents сделать нестатическим, то нужно будет создать объект класса MainForm в классе Program. Типа так:
![image](https://user-images.githubusercontent.com/63415775/81670217-7eee8e80-9471-11ea-80e1-9d70792740d9.png)
И обращаться к нему в программе придется, например, так:
![image](https://user-images.githubusercontent.com/63415775/81670384-b5c4a480-9471-11ea-8d79-6a0804258e6a.png)
Я правильно понимаю?

iKalentii replied:
Нет, просто надо 
`internal SortableBindingList<RadioComponentBase> radioComponents = new SortableBindingList<RadioComponentBase>()`

iKalentii commented on Lab1View/MainForm.cs:
```diff
@@ -1,28 +1,263 @@
+        /// <summary>
+        /// Форма добавления новых радиокомпонентов
+        /// </summary>
+        private readonly AddRadioComponentForm addRadioComponentForm;
```
RSDN стандарт

iKalentii replied:
Подумай- насколько есть смысл хранить его в качестве поля формы. Может быть есть смысл создавать в момент нажатия кнопки

iKalentii commented on Lab1View/MainForm.cs:
```diff
@@ -1,28 +1,263 @@
+            addRadioComponentForm = new AddRadioComponentForm();
+            radioComponents = new SortableBindingList<RadioComponentBase>();
+
+            /// <summary>
```
XML убрать из метода.

iKalentii commented on Lab1View/MainForm.cs:
```diff
@@ -1,28 +1,263 @@
+                Math.Abs(number.Imaginary).ToString(format);
+
+            if (double.IsInfinity(number.Real))
+                realString = infinityString;
```
Скобочки

iKalentii commented on Lab1View/SortableBindingList.cs:
```diff
@@ -0,0 +1,41 @@
+            var sortedList = new List<T>();
+
+            if (direction == ListSortDirection.Ascending)
+                sortedList = originalList.
```
Скобочки

iKalentii commented on Lab1View/SortableBindingList.cs:
```diff
@@ -0,0 +1,41 @@
+                    OrderBy(propertyValue => prop.GetValue(propertyValue)).
+                    ToList();
+            else
+                sortedList = originalList.
```
Скобочки

iKalentii commented on PositiveDoubleTextBox/PositiveDoubleTextBox.cs:
```diff
@@ -0,0 +1,111 @@
+        /// </summary>
+        public PositiveDoubleTextBox()
+        {
+            /// <summary>
```
XML убрать из метода.

iKalentii commented on Lab1View/AddRadioComponentForm.cs:
```diff
@@ -1,21 +1,151 @@
     {
+        /// <summary>
+        /// Генератор случайных положительных целых чисел
+        /// <see cref="int"/>
```
Инт точно не будет переименован, поэтому просто "int" или "целых чисел" хватило бы =)

iKalentii commented on Lab1View/AddRadioComponentForm.cs:
```diff
@@ -1,21 +1,151 @@
+        /// Генератор случайных положительных целых чисел
+        /// <see cref="int"/>
+        /// </summary>
+        private readonly Random randomIntGenerator = new Random();
```
Рандомный генератор не имеет смысла хранить на уровне формы, т.к. это отладочная функция, которая директивами будет вырезаться. Можно запихнуть прям в метод, где вызывается.

iKalentii commented on Lab1View/AddRadioComponentForm.cs:
```diff
@@ -1,21 +1,151 @@
+        /// <see cref="inductorRadioButton"/> или
+        /// <see cref="capacitorRadioButton"/>.
+        /// В качестве параметра в конструктор объекта передается
+        /// <see cref="radioComponentValue"/>
```
Не соответствует сигнатуре метода.

iKalentii commented on Lab1View/MainForm.cs:
```diff
@@ -1,28 +1,263 @@
+    /// <summary>
+    /// Главная форма программы. Также является стартовой формой программы
+    /// </summary>
     public partial class MainForm : Form
```
Зафиксируй масштабирование форм.
![image](https://user-images.githubusercontent.com/902806/81643861-bef14980-9450-11ea-9a1e-9283d0393091.png)

iKalentii commented on Lab1View/MainForm.cs:
```diff
@@ -1,28 +1,263 @@
+    /// <summary>
+    /// Главная форма программы. Также является стартовой формой программы
+    /// </summary>
     public partial class MainForm : Form
```
Бага 
![image](https://user-images.githubusercontent.com/902806/81643955-f4963280-9450-11ea-9bec-2db3d3aab3f4.png)

iKalentii replied:
У меня английская винда, если что

TemZZZ replied:
На счет точки понял. А что с INF не так? Частота же 0 Гц.

iKalentii replied:
Да, всё правильно. Я тупанул, что частота = 0. С INF всё ок.

Commit:
feature(Lab1Model): добавлен класс RadioComponentIO

Commit:
style(RadioComponentBase): добавлены фигурные скобки к блокам if

Commit:
style(AddRadioComponentForm): добавлены фигурные скобки в блоках if

Commit:
style(MainForm): добавлены фигурные скобки в блоках if

Commit:
style(SortableBindingList): добавлены фигурные скобки в блоках if и foreach

Commit:
style(RadioComponentBase): удалены очевидные комментарии

- в методе set свойства Value
- в методе GetImpedance

Commit:
feature(MainForm): удалены комментарии

- очевидный XML-комментарий из конструктора
- в методе DeleteRadioComponentButton_Click XML-комментарии заменены обычными

Commit:
style(PositiveDoubleTextBox): XML-комментарий в конструкторе переделан в простой

Commit:
style(AddRadioComponentForm.AddRadioComponentButton_Click): исправлен комментарий

Commit:
feature(MainForm): установлен фиксированный минимальный размер формы

Commit:
feature(AddRadioComponentForm): генератор случайных чисел randomIntGenerator перенесен в метод GenerateRandomDataButton_Click

Commit:
feature(MainForm): форма addRadioComponentForm теперь создается только при нажатии кнопки addRadioComponentButton

Commit:
feature(SortableBindingList): изменен уровень доступности с internal на public

Commit:
feature(MainForm.radioComponents): изменен уровень доступности поля с internal на public

Commit:
refactor(MainForm): RadioComponents переименован из radioComponents

Commit:
feature(MainForm.RadioComponents): удалено ключевое слово static

Commit:
feature(MainForm): коллекция RadioComponents объявлена как internal и инициализируется сразу

- инициализация RadioComponents  из конструктора класса удалена

Commit:
feature(RadioComponentBase.cs): добавлено событие создания нового радиокомпонента

- добавлен делегат RadioComponentCreatedEventHandler
- добавлен и реализован класс данных события RadioComponentCreatedEventArgs
- в класс AddRadioComponentForm добавлено поле события RadioComponentCreated
- в метод AddRadioComponentButton_Click добавлен вызов события RadioComponentCreated с передачей созданного радиокомпонента

Commit:
feature(MainForm): реализован обработчик события создания нового радиокомпонента OnRadioComponentCreated

feature(MainForm.AddRadioComponentButton_Click): для события addRadioComponentForm.RadioComponentCreated зарегистрирован обработчик OnRadioComponentCreated

Commit:
style(Capacitor): исправлены комментарии у конструкторов

style(Capacitor): добавлены inheritdoc-комментарии к свойствам и к методу CalcImpedance

Commit:
style(Inductor): исправлены комментарии у конструкторов

style(Inductor): добавлены inheritdoc комментарии свойствам и методу CalcImpedance

Commit:
style(RadioComponentBase): исправлен комментарий у конструктора

style(RadioComponentBase): добавлены inheritdoc комментарии свойствам

Commit:
style(Resistor): исправлены комментарии у конструкторов

style(Resistor): добавлены inheritdoc комментарии свойствам и методу CalcImpedance

Commit:
style(AddRadioComponentForm): исправлены/добавлены комментарии

- добавлен комментарий к CancelButton_Click
- исправлен комментарий к AddRadioComponentButton_Click

Commit:
style(MainForm): исправлен комментарий к AddRadioComponentButton_Click

Commit:
style(SortableBindingList): добавлены inheritdoc комментарии

Commit:
fix(PositiveDoubleTextBox): исправлена ошибка некорректного преобразования строк в число из-за регионального стандарта

- реализован метод DotAndCommaToNumberDecimalSeparator, заменяющий точки и запятые на региональный десятичный разделитель
- метод ToPositiveDouble исправлен с учетом корректировки строки методом DotAndCommaToNumberDecimalSeparator

Commit:
style(Capacitor.CalcImpedance): действительная часть комплексного числа переписана через константу zeroRealPart

Commit:
style(Inductor.CalcImpedance): действительная часть комплексного числа переписана через константу zeroRealPart

Commit:
style(ConsoleLoader): удалены комментарии из методов, в блоках if добавлены фигурные скобки

Commit:
style(ConsoleLoader): удалена пустая строка между using

Commit:
style(Main): удален очевидный комментарий

Commit:
style(Main): код оформлен по RSDN

Commit:
feature(ConsoleLoader): добавлен метод DotAndCommaToNumberDecimalSeparator

Commit:
feature(StringToDouble): метод StringToDouble переписан с учетом DotAndCommaToNumberDecimalSeparator

Commit:
fix(ConsoleLoader): исправлен метод PrintComplex

- неверно выводилась бесконечность методом PrintComplex

Commit:
refactor(ConsoleLoader.GetRadioComponent): переименован из ConsoleLoader.GetComponent

Commit:
refactor(ConsoleLoader.AskRadioComponentValue): переименован из ConsoleLoader.AskComponentValue

Commit:
style(Main): длина строк кода теперь не более 78 символов

Commit:
feature(Main): переписана логика программы

- запрос типа радиокомпонента осуществляется в GetRadioComponentLoop
- запрос значения физической величины осуществляется в GetRadioComponentValueLoop
- запрос частоты осуществляется GetFrequencyLoop

Commit:
feature(RadioComponentIO): переписан класс

- удалены методы WriteXml
- удален метод GetFile
- добавлен метод SerializeAndWriteXml
- добавлен метод GetFileStream
- добавлен метод GetStreamReader
- добавлен метод ReadXmlAndDeserialize

Commit:
refactor(XmlReaderWriter): переименован из RadioComponentIO

Commit:
feature(XmlReaderWriter.cs): файл перенесен из проекта Lab1Model в проект Lab1View

Commit:
feature(XmlReaderWriter.cs): изменено пространство имен с Lab1Model на Lab1View

Commit:
style(AddRadioComponentForm.cs): код класса AddRadioComponentForm поставлен первым в файле для возможности открывания формы в конструкторе

Commit:
feature(Lab1View): добавлена форма выбора параметров сохранения радиокомпонентов SetRadioComponentSaveOptionForm

- добавлено перечисление опций сохранения RadioComponentSaveOption
- добавлено событие подтверждения выбора опции RadioComponentReadyToSaveEventHandler
- реализован класс данных события RadioComponentReadyToSaveEventArgs
- реализовано подтверждения выбора опции сохранения кликом по кнопке OkButton_Click
- реализовано закрытие формы кликом по кнопке CancelButton_Click

Commit:
feature(MainForm): добавлена возможность сохранения радиокомпонентов

- метод SaveToFileButton_Click открывает окно для выбора опций сохранения
- метод SaveToFile сохраняет все или выделенную часть компонентов в файл

Commit:
feature(XmlReaderWriter): класс переделан в нестатический

feature(MainForm.SaveToFile): для вызова методов XmlReaderWriter создан объект xmlWriter

Commit:
feature(SetRadioComponentSaveOptionForm.OkButton_Click): после события RadioComponentReadyToSave форма закрывается

Commit:
style(MainForm): добавлены комментарии

- добавлены комментарии к методу GetRadioComponentsToSave
- добавлены комментарии к методу SaveToFile
- добавлены комментарии к методу SaveToFileButton_Click

Commit:
style(SetRadioComponentSaveOptionForm): изменен комментарий к событию RadioComponentReadyToSave

Commit:
feature(XmlReaderWriter): добавлено событие FileSaved

- добавлено возникновение события FileSaved в методе SerializeAndWriteXml

Commit:
feature(XmlReaderWriter): удалено событие FileSaved

Commit:
feature(Lab1View): добавлена форма выбора параметров загрузки радиокомпонентов SetRadioComponentLoadOptionForm

- добавлено перечисление опций загрузки RadioComponentLoadOption
- добавлено событие подтверждения выбора опции RadioComponentReadyToLoadEventHandler
- реализован класс данных события RadioComponentReadyToLoadEventArgs
- реализовано подтверждение выбора опции загрузки кликом по кнопке OkButton_Click
- реализовано закрытие формы кликом по кнопке CancelButton_Click

Commit:
feature(MainForm): добавлена возможность загрузки радиокомпонентов

- метод LoadFromFileButton_Click открывает окно для выбора опций загрузки
- метод LoadFromFile добавляет загруженные радиокомпоненты в конец таблицы или заменяет всю таблицу новыми элементами

Commit:
feature(MainForm): добавлен метод SetupFileDialogs для редактирования внешнего вида окон загрузки и сохранения файлов

- в конструкторе класса метод SetupFileDialogs вызывается

Commit:
style(MainForm): удалены неиспользованные using

Commit:
style(SetRadioComponentLoadOptionForm): зафиксирован размер формы

Commit:
style(SetRadioComponentLoadOptionForm): деактивирована кнопка разворачивания формы на весь экран

Commit:
feature(AddRadioComponentForm): значение resistorRadioButton.Checked = true по умолчанию задано в конструкторе форм

- из конструктора класса установка значения по умолчанию resistorRadioButton.Checked = true удалена

Commit:
style(PositiveDoubleTextBox.Messager): исправлен комментарий

Commit:
feature(MainForm): PositiveDoubleTextBox.Messager определен как поле-делегат ErrorMessager

- в методе SaveToFile вызов PositiveDoubleTextBox.Messager заменен на ErrorMessager
- в методе SaveToFileButton_Click вызов PositiveDoubleTextBox.Messager заменен на ErrorMessager
- в методе LoadFromFile вызов PositiveDoubleTextBox.Messager заменен на ErrorMessager

Commit:
style(MainForm): Text изменен с "MainForm" на "Коллекция радиокомпонентов"

Commit:
feature(AddRadioComponentForm): возвращена установка значения по умолчанию resistorRadioButton.Checked = true в конструктор класса

- в конструкторе форм значение resistorRadioButton.Checked установлено в false

Commit:
fix(PositiveDoubleTextBox): исправлена ошибка - переопределенный метод OnLostFocus блокировал добавление новых обработчиков события LostFocus

- удалено ключевое слово override у OnLostFocus
- изменен уровень доступности с protected на private у OnLostFocus
- в конструкторе класса обработчик OnLostFocus добавляется к событию LostFocus как this.LostFocus += OnLostFocus

Commit:
feature(MainForm): изменена функциональность

- в методе DeleteRadioComponentButton_Click остался только цикл удаления выделенных строк
- метод RadioComponentsDataGridView_RowEnter удален
- функциональность метода RadioComponentsDataGridView_RowEnter перенесена в добавленный метод RadioComponentsDataGridView_SelectionChanged

Commit:
feature(MainForm.FrequencyPositiveDoubleTextBox_LostFocus): изменена функциональность

- теперь вместо обновления radioComponentsDataGridView вызывается RadioComponentsDataGridView_SelectionChanged пересчитывающий импеданс выделенного радиокомпонента
- в конструкторе MainForm событию frequencyPositiveDoubleTextBox.LostFocus добавлен обработчик FrequencyPositiveDoubleTextBox_LostFocus;

Commit:
feature(Lab1View): добавлена форма поиска радиокомпонентов SearchRadioComponentForm

Commit:
feature(MainForm): реализовано создание и открывание формы поиска радиокомпонентов кликом по кнопке SearchButton_Click

Commit:
style(PositiveDoubleTextBox): удалены указатели this

feature(PositiveDoubleTextBox): добавлена константа _defaultText

- элемент по умолчанию при создании заполняется текстом _defaultText
- если после редактирования элемент пустой, то он заполняется текстом _defaultText в методе OnLostFocus

Commit:
feature(SearchRadioComponentForm): реализовано заполнение комбобокса типами радиокомпонентов

feature(SearchRadioComponentForm): реализовано закрывание формы кликом по кнопке CancelButton_Click

Commit:
feature(MainForm): реализовано открывание формы поиска кнопкой SearchButton_Click

Commit:
feature(Lab1View): добавлена форма поиска радиокомпонентов SearchRadioComponentForm

- реализовано закрывание формы кнопкой CancelButton_Click
- реализовано заполнение комбобокса типами радиокомпонентов
- реализовано информирование пользователя об изменении фильтра поиска

Commit:
feature(MinForm): список радиокомпонентов инкапсулирован

- создано приватное поле _radioComponents
- доступ к списку осуществляется через свойство RadioComponents

Commit:
style(AddRadioComponentForm): удален тип  делегата RadioComponentCreatedEventHandler

- событие public event RadioComponentCreatedEventHandler RadioComponentCreated переписано как public event EventHandler<RadioComponentCreatedEventArgs> RadioComponentCreated

Commit:
style(AddRadioComponentForm): класс RadioComponentCreatedEventArgs перенесен в файл RadioComponentCreatedEventArgs.cs

Commit:
feature(SearchRadioComponentForm): свойство RadioComponents переделано в автосвойство с единственным методом get

feature(SearchRadioComponentForm): конструктор переделан в конструктор с параметром

- теперь в конструктор обязательно нужно передать список радиокомпонентов radioComponents
- в конструкторе свойству RadioComponents присваивается параметр radioComponents

feature(SearchRadioComponentForm): реализована реакция на изменения в списке радиокомпонентов в методе OnRadioComponentsChanged

- событию RadioComponents.ListChanged добавлен обработчик OnRadioComponentsChanged в конструкторе класса

Commit:
feature(MainForm): изменено свойство RadioComponents

- свойство RadioComponents переделано в автосвойство
- модификатор доступа изменен на private

feature(MainForm.SearchButton_Click): в конструктор формы searchRadioComponentForm теперь передается параметр RadioComponents

iKalentii commented on Lab1Model_DotNetFramework472/Lab1Model/Lab1Model/Inductor.cs:
```diff
@@ -0,0 +1,45 @@
+			private const string inductorType = "Катушка индуктивности";
+			private const string inductorQuantity = "Индуктивность";
+			/// <inheritdoc/>
+			public override string Unit { get => inductorUnit; }
```
Тут можешь сразу делать Unit => "Гн";  без заведения констант

iKalentii commented on Lab1View/AddRadioComponentForm.cs:
```diff
@@ -1,21 +1,200 @@
+    /// <summary>
+    /// Класс данных события <see cref="RadioComponentCreatedEventHandler"/>
+    /// </summary>
+    public class RadioComponentCreatedEventArgs : EventArgs
```
В отдельный файл.

iKalentii commented on Lab1View/AddRadioComponentForm.cs:
```diff
@@ -1,21 +1,200 @@
+    /// </summary>
+    /// <param name="sender"></param>
+    /// <param name="e"></param>
+    public delegate void RadioComponentCreatedEventHandler(
```
Вместо создания своего делегата корректнее использовать стандартную конструкцию EventHandler<RadioComponentCreatedEventArgs>

Commit:
feature(SearchRadioComponentForm): добавлен метод GetFilteredRadioComponentsIndices

- GetFilteredRadioComponentsIndices возвращающает индексы радиокомпонентов, удовлетворяющих условиям поиска

iKalentii commented on Lab1View/MainForm.cs:
```diff
@@ -1,28 +1,391 @@
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void FrequencyPositiveDoubleTextBox_LostFocus(
```
Тут мог не вызывать через этот обработчик, а сразу подписать событие на обработчик RadioComponentsDataGridView_SelectionChanged

iKalentii commented on Lab1View/SetRadioComponentLoadOptionForm.cs:
```diff
@@ -0,0 +1,105 @@
+		/// Событие, возникающее после подтверждения выбора
+		/// параметра загрузки радиокомпонента
+		/// </summary>
+		public event RadioComponentReadyToLoadEventHandler
```
Выше писал про стандартный EventHandler

Commit:
feature(SearchRadioComponentForm.OnRadioComponentsChanged): добавлена проверка на пустой список

- если список пустой, то кнопка поиска деактивируется и пользователю сообщается о невозможности поиска

iKalentii commented on Lab1View/SetRadioComponentSaveOptionForm.cs:
```diff
@@ -0,0 +1,104 @@
+		/// Событие, возникающее после подтверждения выбора
+		/// параметра сохранения радиокомпонента
+		/// </summary>
+		public event RadioComponentReadyToSaveEventHandler
```
Выше писал про стандартный EventHandler

iKalentii commented on Lab1View/SetRadioComponentLoadOptionForm.cs:
```diff
@@ -0,0 +1,105 @@
+	/// Класс данных события
+	/// <see cref="RadioComponentReadyToLoadEventHandler"/>
+	/// </summary>
+	public class RadioComponentReadyToLoadEventArgs : EventArgs
```
Вынести в отдельный файл

iKalentii commented on Lab1View/SetRadioComponentSaveOptionForm.cs:
```diff
@@ -0,0 +1,104 @@
+	/// Класс данных события
+	/// <see cref="RadioComponentReadyToSaveEventHandler"/>
+	/// </summary>
+	public class RadioComponentReadyToSaveEventArgs : EventArgs
```
Вынести в отдельный файл

Commit:
feature(Lab1View): добавлен класс данных события завершения поиска

- добавлен файл SearchFinishedEventArgs.cs
- в файле реализован класс данных события завершения поиска SearchFinishedEventArgs

Commit:
feature(SearchRadioComponentForm): добавлено событие, происходящее после завершения поиска SearchFinished

iKalentii commented on Lab1View/AddRadioComponentForm.cs:
```diff
@@ -1,21 +1,200 @@
+            {
+                radioComponent = new Capacitor(radioComponentValue);
+            }
+            RadioComponentCreated(this,
```
Если не подписаться на событие - тут вылетит исключение. Лучше перед вызовом проверять на null. Когда переделаешь на стандартное событие - будет проще EventHandler?.Invoke(this, ...)

iKalentii commented on Lab1View/SetRadioComponentLoadOptionForm.cs:
```diff
@@ -0,0 +1,105 @@
+			{
+				loadOption = RadioComponentLoadOption.AddToEnd;
+			}
+			RadioComponentReadyToLoad(this,
```
Если не подписаться на событие - тут вылетит исключение. Лучше перед вызовом проверять на null. Когда переделаешь на стандартное событие - будет проще EventHandler?.Invoke(this, ...)

iKalentii commented on Lab1View/SetRadioComponentSaveOptionForm.cs:
```diff
@@ -0,0 +1,104 @@
+			{
+				saveOption = RadioComponentSaveOption.SaveAll;
+			}
+			RadioComponentReadyToSave(this,
```
Если не подписаться на событие - тут вылетит исключение. Лучше перед вызовом проверять на null. Когда переделаешь на стандартное событие - будет проще EventHandler?.Invoke(this, ...)

Commit:
feature(SearchRadioComponentForm): реализован поиск кликом по кнопке SearchRadioComponentsButton_Click

Commit:
feature(MainForm): реализовано выделение найденых радиокомпонентов OnSearchFinished

Commit:
fix(RegexTextBox.OnTextChanged): переопределенный OnTextChanged не позволял обрабатывать событи TextChanged другим подключенным обработчикам

- изменена сигнатура метода с protected override ... на private ...
- удален вызов base.OnTextChanged(e), если _regex is null
- в конструкторе RegexTextBox к событию TextChanged добавляется обработчик OnTextChanged

Commit:
style(RegexTextBox.OnTextChanged): изменет комментарий, в теле метода одиночный return написан без фигурных скобок {}

Commit:
style(ConsoleLoader.DotAndCommaToNumberDecimalSeparator): sep переименован в regionalNumberDecimalSeparator

Commit:
style(ConsoleLoader.DotAndCommaToNumberDecimalSeparator): изменен комментарий

Commit:
style(ConsoleLoader.StringToDouble): в комментарии стандартные типы данных записаны без тэгов

Commit:
style(ConsoleLoader.IsPositive): в комментарии стандартные типы данных записаны без тэгов

Commit:
style(ConsileLoader.cs): дтрективы using пространства имен System записаны в алфавитном порядке

Commit:
style(MainApp.GetRadioComponentLoop): null в комментарии записан без тэгов

Commit:
style(MainApp.GetRadioComponentValueLoop): стандартные типы данных в комментарии записаны без тэгов

Commit:
style(MainApp.GetFrequencyLoop): стандартные типы данных в комментарии записаны без тэгов

Commit:
style(Capacitor): свойства Unit, Type и Quantity сразу возвращают литеральные константы

- константы capacitorUnit, capacitorType и capacitorQuantity удалены

Commit:
style(Inductor): свойства Unit, Type и Quantity сразу возвращают литеральные константы

- константы inductorUnit, inductorType и inductorQuantity удалены

Commit:
style(Resistor): свойства Unit, Type и Quantity сразу возвращают литеральные константы

- константы resistorUnit, resistorType и resistorQuantity удалены

Commit:
style(IRadioComponent.Value): строки комментария теперь не занимают более 78 символов

Commit:
style(PositiveDoubleTextBox.cs): директивы using пространства имен System отсортированы по алфавиту

Commit:
style(PositiveDoubleTextBox.DotAndCommaToNumberDecimalSeparator): изменен комментарий

style(PositiveDoubleTextBox.DotAndCommaToNumberDecimalSeparator): sep переименован в regionalNumberDecimalSeparator

Commit:
style(PositiveDoubleTextBox.OnLostFocus): изменен комментарий

Commit:
feature(AddRadioComponentForm.AddRadioComponentButton_Click): добавлена проверка подписанного обработчика на null перед вызовом события RadioComponentCreated

- событие вызывается RadioComponentCreated?.Invoke(...)

Commit:
style(SearchFinishedEventArgs): изменен комментарий к конструктоу

Commit:
feature(SetRadioComponentLoadOptionForm.OkButton_Click): добавлена проверка подписанного обработчика на null перед вызовом события RadioComponentReadyToLoad

- событие вызывается RadioComponentReadyToLoad?.Invoke(...)

Commit:
feature(SetRadioComponentLoadOptionForm): изменено объявление события RadioComponentReadyToLoad

- событие объявлено как делегат EventHandler<RadioComponentReadyToLoadEventArgs>

Commit:
style(SetRadioComponentLoadOptionForm.cs): исправлен комментарий к классу RadioComponentReadyToLoadEventArgs

Commit:
feature(SetRadioComponentLoadOptionForm.cs): удален делегат RadioComponentReadyToLoadEventHandler

Commit:
feature(SetRadioComponentLoadOptionForm.cs): класс RadioComponentReadyToLoadEventArgs перемещен в файл RadioComponentReadyToLoadEventArgs.cs

Commit:
feature(SetRadioComponentLoadOptionForm.cs): перечисление RadioComponentLoadOption перемещено в файл RadioComponentReadyToLoadEventArgs.cs

Commit:
feature(SetRadioComponentSaveOptionForm): изменено объявление события RadioComponentReadyToSave

- событие объявлено как EventHandler<RadioComponentReadyToSaveEventArgs>

Commit:
feature(SetRadioComponentSaveOptionForm.cs): удален делегат RadioComponentReadyToSaveEventHandler

Commit:
style(SetRadioComponentSaveOptionForm.cs): изменен комментарий к классу RadioComponentReadyToSaveEventArgs

Commit:
style(SetRadioComponentSaveOptionForm.cs): класс RadioComponentReadyToSaveEventArgs перемещен в файл RadioComponentReadyToSaveEventArgs.cs

Commit:
style(SetRadioComponentSaveOptionForm.cs): перечисление RadioComponentSaveOption перемещено в файл RadioComponentReadyToSaveEventArgs.cs

Commit:
feature(SetRadioComponentSaveOptionForm.OkButton_Click): добавлена проверка подписанного обработчика на null перед вызовом события RadioComponentReadyToSave

- событие вызывается RadioComponentReadyToSave?.Invoke(...)

Commit:
style(XmlReaderWriter.cs): директивы using отсортированы по алфавиту

Commit:
feature(MainForm): в конструкторе класса на событие frequencyPositiveDoubleTextBox.LostFocus подписан RadioComponentsDataGridView_SelectionChanged

Commit:
feature(MainForm): удалена подписка обработчика FrequencyPositiveDoubleTextBox_LostFocus на frequencyPositiveDoubleTextBox.LostFocus

Commit:
feature(MainForm): удален метод FrequencyPositiveDoubleTextBox_LostFocus

Commit:
style(MainForm.ComplexToText): исправлен комментарий

Commit:
fix(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): поиск выполнялся неправильно

- если фильтрации меньше, больше, равно давали пустые множества, то фильтр по типу радиокомпонента не сбрасывался
- добавлена проверка включенных фильтров после фильтрации по типу: если ни один фильтр из больше, меньше, равно не включен, то сразу возвращаются индексы отфильтрованных компонентов по типу
- удалена проверка на пустое множество перед последним return

Commit:
style(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): переформатированы переносы длинных инструкций

Commit:
style(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): добавлены комментарии в метод

Commit:
feature(XmlReaderWriter.SerializeAndWriteXml): добавлен отлов исключения SerializeAndWriteXml

- при возникновении исключения SerializeAndWriteXml через messager отправляется сообщение о невозможности сериализации объекта
- при возникновении другого типа исключения через messager отправляется сообщение исключения, и исключение создается повторно

Commit:
feature(XmlReaderWriter.ReadXmlAndDeserialize): добавлен отлов исключения SerializeAndWriteXml

- при возникновении исключения SerializeAndWriteXml через messager отправляется сообщение о невозможности десериализации в объект
- при возникновении другого типа исключения через messager отправляется сообщение исключения, и исключение создается повторно

Commit:
style(XmlReaderWriter.GetFileStream): null в комментарии записан без тэгов

Commit:
style(XmlReaderWriter.GetStreamReader): null в комментарии записан без тэгов

Commit:
style(XmlReaderWriter.ReadXmlAndDeserialize): null в комментарии записан без тэгов

Commit:
feature(Lab1View): добавлен класс вызова функции и обработки возможных исключений ExceptionHandler

- реализован обобщенный метод вызова функции и обработки возможных исключений CallFunction

Commit:
fix(ExceptionHandler.CallFunction): в блоке catch всегда обрабатывается первое исключение из перечислителя exceptionToMessageMap

- в цикле foreach блока catch в операторе сравнения типов исключений if (!(exceptionToMessage.Key.GetType() != e.GetType())) удален оператор !

Commit:
feature(XmlReaderWriter): методы GetFileStream и GetStreamReader переписаны через вызов ExceptionHandler.CallFunction(...)

Commit:
feature(ExceptionHandler.CallFunction): изменен тип входного параметра

- тип параметра exceptionToMessageMap изменен с IEnumerable<KeyValuePair<Exception, string>> на IEnumerable<KeyValuePair<Type, string>>
- имя параметра exceptionToMessageMap изменено на exceptionTypeToMessageMap
- в операторе сравнения в цикле foreach вызов exceptionToMessage.Key.GetType() изменен на exceptionToMessage.Key
- исправлен комментарий

Commit:
refactor(XmlReaderWriter): словарь StreamExceptionToMessageMap переименован в StreamExceptionTypeToMessageMap

feature(XmlReaderWriter): изменен словарь StreamExceptionTypeToMessageMap

- теперь словарь заполняется не парами {Exception obj, string message}, а парами {typeof(<класс исключения>), string message}

Commit:
style(XmlReaderWriter.StreamExceptionTypeToMessageMap): добавлен комментарий

Commit:
feature(XmlReaderWriter.StreamExceptionTypeToMessageMap): изменено сообщение для типа исключения UnauthorizedAccessException

Commit:
feature(XmlReaderWriter.StreamExceptionTypeToMessageMap): добавлен элемент

- добавлен {typeof(IOException), string messageOnIOException}

iKalentii commented on Lab1View/SearchRadioComponentForm.cs:
```diff
@@ -0,0 +1,296 @@
+		/// Список радиокомпонентов, по которым осуществляется поиск
+		/// </summary>
+		private SortableBindingList<RadioComponentBase> RadioComponents
+		{ get; } = new SortableBindingList<RadioComponentBase>();
```
Тут можно инициалцизацию убрать, т.к. у тебя в конструкторе всё равно новая ссылка приходит, и get перенести на нужную строку

TemZZZ replied:
На нужную строку, это как? Вот так могу сделать
![image](https://user-images.githubusercontent.com/63415775/82463445-0b8af380-9ae7-11ea-897a-b34c0d1a53a3.png)
Если так оставить:
private SortableBindingList<RadioComponentBase> RadioComponents
![image](https://user-images.githubusercontent.com/63415775/82464293-1c883480-9ae8-11ea-9b8c-7025715bf047.png)
то там за 78 символов переваливает. Ненамного, но все же.

iKalentii commented on Lab1View/SearchRadioComponentForm.cs:
```diff
@@ -0,0 +1,296 @@
+
+			searchRadioComponentsButton.Enabled = false;
+
+			SearchFinished(this, new SearchFinishedEventArgs(foundIndices));
```
Проверка на null

iKalentii commented on Lab1View/SearchRadioComponentForm.cs:
```diff
@@ -0,0 +1,296 @@
+					.ToArray();
+			}
+
+			// Фильтр меньше, чем
```
Большой метод, я бы фильтры побил в разные методы, которые возвращают тебе результат.

iKalentii commented on Lab1View/SearchRadioComponentForm.cs:
```diff
@@ -0,0 +1,296 @@
+				lessThanIndexToRadioComponentMap.Intersect(
+					moreThanIndexToRadioComponentMap);
+
+			if (filteredIndexToRadioComponentMap.Count() == 0)
```
Тут лучше вызвать Any()
https://stackoverflow.com/questions/305092/which-method-performs-better-any-vs-count-0

iKalentii commented on Lab1View/XmlReaderWriter.cs:
```diff
@@ -0,0 +1,145 @@
+		/// и соответствующие им сообщения
+		/// </summary>
+		private Dictionary<Type, string> StreamExceptionTypeToMessageMap
+		{ get; } = new Dictionary<Type, string>
```
Я, конечно тебе вчера сказал переносить, но моё чувство прекрасного бунтует) Мне привычнее вот так
![image](https://user-images.githubusercontent.com/902806/82435700-e2566d00-9abe-11ea-8f46-a491f5f52355.png)

Но это вкусовщина, поэтому не критично.

iKalentii commented on Lab1View/RadioComponentReadyToSaveEventArgs.cs:
```diff
@@ -0,0 +1,38 @@
+	/// Класс данных события подтверждения выбора
+	/// параметра сохранения радиокомпонента
+	/// </summary>
+	public class RadioComponentReadyToSaveEventArgs : EventArgs
```
По организации решения - все EventArgs можно было бы вынести в отдельную папку решения, чтобы маленько сгрупировать сущности. Также и формы можно было бы собрать во View.

Commit:
feature(SearchRadioComponentForm.SearchRadioComponentsButton_Click): добавлена проверка обработчика на null перед вызовом события SearchFinished

- Вызов события SearchFinished(...) переписан как SearchFinished?.Invoke(...)

Commit:
feature(SearchRadioComponentForm): удалена инициализация списка RadioComponents

Commit:
feature(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): проверка filteredIndexToRadioComponentMap.Count() == 0 заменена на filteredIndexToRadioComponentMap.Any()

Commit:
feature(Lab1View): добавлен класс с методами расширения для фильтрации радиокомпонентов WhereExtension

Commit:
feature(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): метод переписан через функции класса WhereExtension

fix(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): неверно выполнялась фильтрация

- добавлен оператор ! перед проверкой filteredIndexToRadioComponentMap.Any()

Commit:
feature(WhereExtension): добавлен метод ToIndexToRadioComponentMap

- ToIndexToRadioComponentMap возвращает перечислитель "индекс-радиокомпонент" на основе списка радиокомпонентов

Commit:
feature(WhereExtension): добавлен метод GetIndices

- метод GetIndices возвращает индексы перечислителя проиндексированных радиокомпонентов

Commit:
feature(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): метод переписан с использованием методов WhereExtension.ToIndexToRadioComponentMap и WhereExtension.GetIndices

Commit:
style(SearchRadioComponentForm.cs): удалены неиспользуемые директивы using

Commit:
style(MainForm): отформатированы строки инициализации RadioComponents

Commit:
style(SearchRadioComponentForm): отформатированы строки объявления RadioComponents

Commit:
style(Lab1View): все формы перенесены в папку View

Commit:
style(Lab1View): все EventArgs классы перенесены в папку EventArgsClasses

iKalentii commented on Lab1View/EventArgsClasses/RadioComponentCreatedEventArgs.cs:
```diff
@@ -0,0 +1,31 @@
+﻿#define TEST
```
Лишняя директива

iKalentii commented on Lab1View/View/SetRadioComponentSaveOptionForm.cs:
```diff
@@ -0,0 +1,63 @@
+		/// <param name="e"></param>
+		private void OkButton_Click(object sender, EventArgs e)
+		{
+			RadioComponentSaveOption saveOption;
```
Тут можно немного локаничнее написать
`RadioComponentSaveOption saveOption = saveSelectedRadioButton.Checked 
? RadioComponentSaveOption.SaveSelected
: RadioComponentSaveOption.SaveAll`

Commit:
style(RadioComponentCreatedEventArgs.cs): удалена директива TEST

Commit:
style(SetRadioComponentSaveOptionForm.OkButton_Click): выбор опции сохранения переписан через тернарный оператор

Commit:
style(SetRadioComponentLoadOptionForm.OkButton_Click): выбор опции загрузки переписан через тернарный оператор

TemZZZ committed:
Merging: develop <-- console-loader-develop

Слияние ветви console-loader-develop (доработка ConsoleLoader) в ветвью develop
